### PR TITLE
Release 0.1.0 - Add Terraform files

### PR DESCRIPTION
**Pull request by dalenewby**
### Added

 - Added Terraform files

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -0,0 +1,119 @@
+locals {
+  app_name_and_env = "${var.app_name}-${data.terraform_remote_state.common.outputs.app_env}"
+  app_env          = data.terraform_remote_state.common.outputs.app_env
```
Comment (briskt) [inline main.tf:3]: I kind of wonder whether we should deprecate `app_env` and only use the longer version. It seems a little silly to have both.

File: main.tf [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  app_environment  = data.terraform_remote_state.common.outputs.app_environment
+  name_tag_suffix  = "${var.app_name}-${var.customer}-${local.app_environment}"
+}
+
+/*
+ * Create cloudwatch log group for app logs
+ */
+resource "aws_cloudwatch_log_group" "cw_b2_fsbackup" {
+  name              = "${var.app_name}-${var.app_env}"
```
Comment (briskt) [inline main.tf:12]:
```suggestion
  name              = local.app_name_and_env
```

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      restic_host           = var.b2_fsbackup_host
+      repo_pw               = random_id.b2_repo_password.hex
+      repo_string           = var.b2_fsbackup_repo
+      restic_tag            = "${var.app_name}-${var.app_env}-b2-fs-backup"
+      source_path           = var.backup_path
+      cpu                   = var.cpu
+      memory                = var.memory
+      docker_image          = "silintl/tfc-backup-b2"
+      docker_tag            = var.docker_tag
+      #  app_env               = var.app_env
```
Comment (briskt) [inline main.tf:49]: Depending on why this is here:
```suggestion
      #  app_env               = local.app_env
```

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+      aws_region            = var.aws_region
+      aws_access_key_id     = var.aws_access_key
+      aws_secret_access_key = var.aws_secret_key
+      log_level             = var.log_level
+      app_name              = var.app_name
+      cw_log_group          = aws_cloudwatch_log_group.cw_b2_fsbackup.name
+      cw_stream_prefix      = "B2_FS_backup-${var.app_name}-${var.app_env}"
```
Comment (briskt) [inline main.tf:56]:
```suggestion
      cw_stream_prefix      = "B2_FS_backup-${local.app_name_and_env}"
```

File: main.tf [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    }
+  )
+}
+
+resource "aws_ecs_task_definition" "b2_fstd" {
+  family                = "${var.app_name}-b2-fsbackup-${var.app_env}"
```
Comment (briskt) [inline main.tf:62]:
```suggestion
  family                = "${var.app_name}-b2-fsbackup-${local.app_env}"
```

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+  container_definitions = local.task_def
+  #??  task_role_arn      = var.task_role_arn #???
```
Comment (briskt) [inline main.tf:64]: I _think_ you'll need this. I'm a little fuzzy on it, though.

File: main.tf [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+  #??  network_mode       = var.network_mode  #???
+}
+
+/*
+ * Create role for scheduled running of backup task definitions.
+ */
+resource "aws_iam_role" "ecs_events" {
+  name = "ecs_events-${var.app_name}-${var.app_env}"
```
Comment (briskt) [inline main.tf:72]:
```suggestion
  name = "ecs_events-${local.app_name_and_env}"
```

File: main.tf [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+}
+DOC
+
+}
+
+/*
+ * CloudWatch configuration to start file system backup.
+ */
+resource "aws_cloudwatch_event_rule" "b2_fsbackup_rule" {
+  name        = "b2_fsbackup-${var.app_name}-${var.app_env}"
```
Comment (briskt) [inline main.tf:96]:
```suggestion
  name        = "b2_fsbackup-${local.app_name_and_env}"
```

File: main.tf [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+  description = "Start B2 file system backup on cron schedule"
+
+  schedule_expression = "cron(${var.b2_fsbackup_schedule})"
+  is_enabled          = var.b2_fsbackup_enabled
+
+  tags = {
+    app_name = var.app_name
+    app_env  = var.app_env
```
Comment (briskt) [inline main.tf:104]:
```suggestion
    app_env  = local.app_env
```

File: main.tf [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+  }
+}
+
+resource "aws_cloudwatch_event_target" "b2_fsbackup" {
+  target_id = "run-b2-fsbackup-${var.app_name}-${var.app_env}"
```
Comment (briskt) [inline main.tf:109]:
```suggestion
  target_id = "run-b2-fsbackup-${local.app_name_and_env}"
```

File: remote.tf [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+  }
+}
+
+data "terraform_remote_state" "b2-bucket" {
+  backend = "remote"
+
+  config = {
+    organization = "gtis"
+    workspaces = {
+      name = "b2-${var.app_name}-${var.app_env}"
```
Comment (briskt) [inline remote.tf:18]:
```suggestion
      name = "b2-${local.app_name_and_env}"
```

File: vars.tf [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+variable "aws_access_key" {
+}
+
+variable "aws_secret_key" {
+}
+
+variable "app_env" {
+  default = "prod"
+}
+
```
Comment (briskt) [inline vars.tf:14]: `app_env` isn't needed since it's provided by the shared workspace
```suggestion
```

File: vars.tf [hunk 13]
```diff
@@ -0,0 +1,98 @@
+variable "aws_region" {
+  default = "us-east-1"
+}
+
+variable "aws_access_key" {
+}
```
Comment (briskt) [inline vars.tf:6]: This allows for running terraform on the command-line using environment variable `AWS_ACCESS_KEY_ID` to provide the value.
```suggestion
variable "aws_access_key" {
  default = null
}
```

File: vars.tf [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+
+variable "aws_secret_key" {
+}
```
Comment (briskt) [inline vars.tf:9]: (see above)
```suggestion
variable "aws_secret_key" {
  default = null
}
```

File: vars.tf [hunk 15]
```diff
@@ -0,0 +1,98 @@
...
+  default = "tfcbackup"
+}
+
+variable "cpu" {
+  default = "200"
+}
+
+variable "memory" {
+  default = "128"
+}
```
Comment (briskt) [inline vars.tf:25]:
```suggestion
variable "cpu" {
  default = 200
}

variable "memory" {
  default = 128
}
```

File: task-definition-b2-backup.tftpl [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+      {
+        "name": "AWS_ACCESS_KEY_ID",
+        "value": "${aws_access_key_id}"
+      },
+      {
+        "name": "AWS_SECRET_ACCESS_KEY",
+        "value": "${aws_secret_access_key}"
+      },
+      {
+        "name": "LOG_LEVEL",
```
Comment (briskt) [inline task-definition-b2-backup.tftpl:79]: Is this needed? I don't see it in tfc-backup-b2.

File: task-definition-b2-backup.tftpl [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+      }
+    },
+    "entryPoint": null,
+    "portMappings": [],
+    "command": [],
+    "linuxParameters": null,
+    "cpu": ${cpu},
+    "environment":
+      {
+        "name": "ATLAS_TOKEN",
```
Comment (briskt) [inline task-definition-b2-backup.tftpl:19]: Change to `TFE_TOKEN`?

**Review by briskt — COMMENTED** [review]
Some things to consider...

File: main.tf (hunk 2 above)
Comment (dalenewby) [reply main.tf:12]: Good catch, thanks!

File: task-definition-b2-backup.tftpl (hunk 17 above)
Comment (dalenewby) [reply task-definition-b2-backup.tftpl:19]: `ATLAS_TOKEN` is required by `tfc-ops` which is used by `tfc-dump.pl`.

File: task-definition-b2-backup.tftpl (hunk 16 above)
Comment (dalenewby) [reply task-definition-b2-backup.tftpl:79]: I don't think it is needed.  It was a copy/paste artifact.

File: task-definition-b2-backup.tftpl (hunk 17 above)
Comment (briskt) [reply task-definition-b2-backup.tftpl:19]: That is correct. I somehow forgot you were using tfc-ops

**Review by briskt — APPROVED** [review]

**Pull request merged**